### DISCUSSION TITLE
Filter Option for Chapters according to docdegree

### DISCUSSION THREAD

**Discussion by Jmmo-Gomes** [joaomlourenco/novathesis#404] [Q&A] [answered]
## Problem

I want to have a quick way of reusing the chapters common to the proposal and the dissertation thesis and to only use those that are meant for each one by only changing the docdegree option in 1_novathesis.tex.

## Solution

As I cant use the current implementation of the command \ifdocclass in novathesis.cls for this effect and adding a new entry for the list would break the cover layout, I made up my own solution that works and it is the following:

### New Latex Commands in novathesis.cls
![novathesis cls](https://user-images.githubusercontent.com/72549192/211809584-76da9ee0-c1e1-442c-be7b-10c33f71c1cc.PNG)

### Usage of Latex Comands in 4_files.tex
![4_files tex](https://user-images.githubusercontent.com/72549192/211809539-0cad83b4-8019-4d85-81ea-cb2007e0c21a.PNG)

Is there a better/smart way to archieve the desired effect?

**Comment by joaomlourenco** [comment, marked as answer]
@Jmmo-Gomes 
Thank you for the interesting question/challenge.  And most specially for coming up with a solution by yourself. 👍 

Right now I cannot think of a better solution than the one you devised.  But, still, I have two comments on your solution:
1. I **strongly discourage** people from changing `template.tex`, `novathesis.cls`, and any other file in folder `NOVAtehsisFiles`.  If you do, you cannot (easily) upgrade to newer versions anymore.  I suggest you add them in the file `Config/5_packages.tex`.
2. Your new commands _down there_ depend on `\ifoptionequal`, and that command accepts 4 arguments: _compare1_, compare2, _true branch_, _false branch_.  So, you your case, you want to add an extra `{}` in the end for the false branch, i.e.,
```latex
% Proposal Chapters
\ifmscplandoc{
Intaddfile{chapter}{chapters/Proposal/4-Proposal/chapter 4}
Intaddfile{chapter}{chapters/Proposal/5-Proof Of Concept/chapter5}
Intaddfile{chapter}{chapters/Proposal/6-Work Plan/chapter6}
}{}%Add {} here for the false branch!!!!

% Thesis Chapters
\ifmscthesisdoc{
Intaddfile{chapter}{chapters/Thesis/4-Project Plan/chapter4}
...
Intaddfile{chapter}{chapters/Thesis/12-Future Work/chapter12}
}{}%Add {} here for the false branch!!!!
```

**Comment by joaomlourenco** [comment]
@Jmmo-Gomes As a side-note, I'm preparing version 7.0.0 of the template and, in this version, this feature will be available as… 
```latex
\ntaddfile{chapter}{my-chapter} % for all documents
\ntaddfile{chapter}[phd]{my-chapter} % just for phd document
\ntaddfile{chapter}[phd,phdplan]{my-chapter} % for both phd and phdplan documents
```
